test/unit/soroban_test.js

```diff
@@ -0,0 +1,43 @@
+describe('Soroban', function () {
+  describe('formatTokenAmount', function () {
+    it("returns '100.0000001' for '1000000001' of a token set to 7 decimals", function () {
+      const amount = 1000000001;
+
+      expect(StellarBase.Soroban.formatTokenAmount(amount, 7)).to.equal(
+        '100.0000001'
+      );
+    });
+
+    it("returns '100000.0001' for '10000000010' of a token set to 5 decimals", function () {
+      const amount = '10000000010';
+
+      expect(StellarBase.Soroban.formatTokenAmount(amount, 5)).to.equal(
+        '100000.0001'
+      );
+    });
+
+    it("returns '10000.00001' for '1000000001.' of a token set to 5 decimals", function () {
+      const amount = '1000000001.';
+
+      expect(StellarBase.Soroban.formatTokenAmount(amount, 5)).to.equal(
+        '10000.00001'
+      );
+    });
```

Comment:
these tests would be more readable and extensible as a loop over an array of test cases, like

```js
[
  [ 1000000001, 7, '100.0000000' ],
  // etc.
].forEach(([ amount, decimals, expected ]) => {
});
```

Reply:
i'm so noob here LOL